nf_core/lint.py

```diff
@@ -286,6 +291,57 @@ def check_readme(self):
                 self.failed.append((6, "Found a bioconda environment.yml file but no badge in the README"))
 
 
+    def check_version_consistency(self):
+        """ Checking if versions are consistent between container,
+        release tag and config and numeric """
+
+        logging.debug('Checking if versions are consistent between container, release tag and config and numeric')
+        versions = {}
+        # Get the version definitions
+        # Get version from nextflow.config
+        versions['pipeline_version'] = self.config['params.version'].strip()
+
+        # Get version from the docker slug
+        if self.config.get('params.container') and \
+                not ':' in self.config['params.container']:
+            self.failed.append((7, "Docker slug seems not to have "
+                "a version tag: {}".format(self.config['params.container'])))
+            return
+        # Avoid key error, as this parameter is not mandatory
+        if self.config.get('params.container'):
+            versions['container_version'] = self.config['params.container'].strip().split(':')[-1]
+
+        # Check, if the process.container variable is set properly
+        if self.config.get('process.container') and \
+                not self.config['process.container'] == 'params.container':
```

Comment:
What do you think about simply checking that it does == `params.container`? I can't think of a situation where it would make sense to have both defined but set to different values..

Reply:
Exactly, it does not make sense and this is why I want to make sure it isn't :)

Reply:
But of course if you like we can make it mandatory to have `== params.config` defined here, I have no hard feelings about that ;)

Reply:
I think let's just check that they're equal - keeps things simple, plus it'll improve the test coverage 😉

Reply:
okok :D

Reply:
Quick question, probably a little late, but what happen when a pipeline has more than one container?

Reply:
Then `process.container` won't be specified and this test won't run. There will be a warning (but not a fail) in an earlier test if `process.container` is not specified.